### PR TITLE
Fixed year handling in macOS date formats

### PR DESCRIPTION
**Pull request by lenlo**
I have set the system date format to yyyy-MM-dd on my Mac, but when I started using moneyGuru, I found that it would produce dates of the form dd-MM-yyyy instead. The reason for this is that macOS (and possibly others) use a single 'y' to denote a non-zerofilled complete year. moneyGuru didn't recognize this and defaulted to its fallback format instead.

Fixed by substituting 'yyyy' for 'y' before parsing the format string. This will give us zero filled centuries for years < 1000, but that's about as close as we can get giving strftime's limitations. Hopefully, there's no real need to process historic accounting records for Roman times anyway.

### CODE REVIEW

File: core/model/date.py [hunk 1]
```diff
@@ -606,3 +609,5 @@ def sys_format(self):
         repl_elems = [self.ISO2SYS[elem] for elem in self.elements]
         return self.separator.join(repl_elems)
 
+# Default user format
+DateFormat.user = DateFormat(None)
```
Comment (hsoft) [inline core/model/date.py]: What is this line supposed to do?

Comment (lenlo) [inline core/model/date.py]: Oh. That's just a leftover that should have gotten removed. I'll do that right away.

**Comment by lenlo** [comment]
Feel free to add a test case. I'm not very familiar with how the unit tests work.

**Comment by hsoft** [comment]
I don't know when I'll have time for this myself, but in any case, I'll be glad to merge this as soon as it has unit tests.

**Comment by lenlo** [comment]
Added a test case that I hope is appropriate.

**Pull request merged**